tmva/sofie_parsers/src/RModelParser_ONNX.cxx

```diff
@@ -1216,6 +1216,39 @@ std::unique_ptr<ROperator> make_ROperator_Concat(const onnx::NodeProto &nodeprot
    return op;
 }
 
+std::unique_ptr<ROperator> make_ROperator_Shape(const onnx::NodeProto& nodeproto, const onnx::GraphProto& /*graphproto */, std::unordered_map<std::string, ETensorType>& tensor_type){
+
+   ETensorType input_type;
+
+   auto input_name =  nodeproto.input(0);
+   auto it = tensor_type.find(input_name);
+   if (it != tensor_type.end()){
+      input_type = it->second;
+   }else{
+      throw std::runtime_error("TMVA::SOFIE ONNX Parser Shape op has input tensor" + input_name + " but its type is not yet registered");
+   }
+
+   std::unique_ptr<ROperator> op;
+
+
+   switch(input_type){
+   case ETensorType::FLOAT:
```

Comment:
The logic here is not correct. We can support input types FLOAT, DOUBLE and maybe also integer. 
The input types can then be the template parameter of the ROperator_Shape.
The output type is fixed to be INT64 according to ONNX specifications